Objects/typeobject.c

```diff
@@ -2990,11 +2989,10 @@ PyType_FromModuleAndSpec(PyObject *module, PyType_Spec *spec, PyObject *bases)
     type->tp_as_sequence = &res->as_sequence;
     type->tp_as_mapping = &res->as_mapping;
     type->tp_as_buffer = &res->as_buffer;
-    /* Set tp_base and tp_bases */
+    /* Set tp_bases and tp_base */
     type->tp_bases = bases;
     bases = NULL;
-    Py_INCREF(base);
-    type->tp_base = base;
+    PyType_SetBase(type, base);
```

Comment:
If Line 2995 can be replaced, why Line 842 can not do the same operation?
https://github.com/python/cpython/blob/master/Objects/typeobject.c#L842

Reply:
type_set_bases() is very complex, I prefer to leave it unchanged to reduce the risk of any regression. IMHO the current type_set_bases() code is simpler to follow without an indirection like PyType_SetBase() function call.